pkgs/development/python-modules/cfg_diag/default.nix

```diff
@@ -0,0 +1,44 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+, pytest
+, python
+, setuptools
+}:
+
+let
+  cfg_diag = buildPythonPackage rec {
+    pname = "cfg_diag";
+    version = "0.4.0";
+    format = "pyproject";
+
+    src = fetchFromGitHub {
+      owner = "storpool";
+      repo = "python-cfg_diag";
+      rev = "release/${version}";
+      sha256 = "y9g/0JzbeF1gab/HaTfKj1AtAdcY6LO1aeEn+A2Ms/s=";
+    };
+
+    meta = with lib; {
+      homepage = "https://github.com/storpool/python-cfg_diag";
+      description = "A common configuration-storage class with a .diag() method";
+      license = licenses.bsd2;
+      maintainers = with maintainers; [ ppentchev ];
+    };
+
+    propagatedBuildInputs = [
+      setuptools
+    ];
+
+    checkInputs = [
+      pytest
+    ];
+
+    checkPhase = ''
+      runHook preCheck
+      PYTHONPATH="''${PYTHONPATH+''$PYTHONPATH:}$(pwd)/src" ${python.interpreter} -m pytest -v unit_tests
+      runHook postCheck
+    '';
```

Comment:
This isn't really how tests are done for python packages. Did you try just adding `pytestCheckHook` to `checkInputs` and seeing if its default `checkPhase` works?

Specifically, one issue with this `checkPhase` is that it tests the non-installed source. `buildPythonPackage` actually redirects all `check*`-related attributes to `installCheck*` ones, and tests for python packages are supposed to test the _installed_ code.

`pytestCheckHook` attempts to do this automatically and works in most cases.

Reply:
Yeah, well, my bad, thanks a lot for pointing this out! The truth is, I did notice the comments about check actually being installCheck when looking at the source of the functions I was using, but I filed it as "yeah, need to look into that later" and did not quite grasp the meaning behind it. It does make a lot of sense.

And, TBH, I have absolutely no idea how I missed pytestCheckHook when looking at other Python packages :/

Thanks a lot, fixed all of these by using pytestCheckHook and removing the manual checkPhase overrides.